web/src/engine/websites/decorators/Grouple.ts

```diff
@@ -0,0 +1,111 @@
+import { Fetch, FetchWindowScript } from '../../platform/FetchProvider';
+import { type MangaScraper, type Chapter, Page } from '../../providers/MangaPlugin';
+import { type Priority } from '../../taskpool/DeferredTask';
+import * as Common from './Common';
+
+const pagesWithServersScript = `
+    new Promise(resolve => {
+        const pics = rm_h.pics.map(pic => {
+            const picUrl = new URL(rm_h.reader.preparePicUrl(pic.url.replace(/^\\/\\//, 'https://')));
+            return picUrl.pathname + picUrl.search;
+        });
+        const servers =  rm_h.servers.map(server => server.path.replace(/^\\/\\//, 'https://'));
+        const mainserver = servers.shift();
+        resolve( pics.map( picture => {
+            const mirrors = servers.map(server => new URL(picture, server).href);
+            return { url : new URL(picture, mainserver).href, mirrors };
+        }));
+       resolve(pics);
+    });
+`;
+
+export const chapterScript = `
+    new Promise(resolve => {
+        resolve([...document.querySelectorAll('a.chapter-link.cp-l')].map(chapter => {
+            return {
+                id: chapter.pathname + chapter.search,
+                title: chapter.text.trim(),
+            };
+        }));
+    });
+`;
+
+export const queryMangas = 'div.tile div.desc h3 a';
+export const queryMangaTitle = 'meta[itemprop = "name"]';
+export const pathMangas = '/list?offset={page}';
+export const pageMangaOffset = 70;
+export const queryPages = [
+    'div#all img.img-responsive',
+    'div.text-center img[loading="lazy"]',
+].join(',');
+
+type ImagesData = {
+    url: string,
+    mirrors: string[];
+}
+
+type PageParameters = {
+    mirrors: string[],
+};
+
+/*************************************************
+ ******** Pages Extraction Methods ********
+ *************************************************/
+
+/**
+ * Extracts the pages from the HTML page of the given {@link chapter}.
+ * @param this - A reference to the {@link MangaScraper}
+ * @param chapter - A reference to the {@link Chapter} which contains the pages
+ * @param script - A script to extract the pages as \{ url : string, mirrors: string[] \}
+ */
+async function FetchPagesSinglePageJS(this: MangaScraper, chapter: Chapter, script: string = pagesWithServersScript, delay: number = 1500): Promise<Page[]> {
```

Comment:
This method has a small footprint, is not exported and only used once.
Implementation might as well be merged directly into `PagesSinglePageJS`.

Reply:
You think we should alter Chapter gathering to add the parameter in script and alter FetchPagesSinglePageJS to handle mirrors?